src/notation/internal/notationinteraction.cpp

```diff
@@ -7339,6 +7339,57 @@ void NotationInteraction::addMelisma()
     toLyrics->selectAll(toLyrics->cursor());
 }
 
+void NotationInteraction::addElision() {
+
+    if (!m_editData.element || !m_editData.element->isLyrics()) {
+        LOGW("addElision called outside of lyrics editing context.");
+        return;
+    }
+
+    // fetch the text‐edit data & cursor
+    auto edd = m_editData.getData(m_editData.element);
+    if (!edd) {
+        LOGE("addElision: Failed to get EditData for the element.");
+        return;
+    }
+    auto* ted = static_cast<TextEditData*>(edd.get());
+    if (!ted) {
+        LOGE("addElision: EditData is not TextEditData.");
+        return;
+    }
```

Comment:
`static_cast` (in contrast to `dynamic_cast`) doesn't change anything about whether it's `nullptr` or not, so only one null check would be necessary; and judging from other similar places in the code, we can safely assume that these things are not nullptr. So:
```suggestion
    TextEditData* ted = static_cast<TextEditData*>(m_editData.getData(m_editData.element).get());
```